src/client/api/rpc.c

```diff
@@ -64,3 +64,56 @@ daos_rpc_send(crt_rpc_t *rpc, tse_task_t *task)
 
 	return rc;
 }
+
+struct daos_rpc_status {
+	int completed;
+	int status;
+};
+
+static int
+rpc_progress_cb(void *arg)
+{
+	struct daos_rpc_status *status = arg;
+
+	return status->completed;
+}
+
+static void
+daos_rpc_wait(crt_context_t *ctx, struct daos_rpc_status *status)
+{
+	/* Wait on the event to complete */
+	while (!status->completed) {
+		int rc = 0;
+
+		rc = crt_progress(ctx, DAOS_EQ_WAIT,
+				  rpc_progress_cb, status);
+		if (rc)
+			break;
+	}
+}
+
+static void
+daos_rpc_wait_cb(const struct crt_cb_info *cb_info)
+{
+	struct daos_rpc_status *status = cb_info->cci_arg;
+
+	status->completed = 1;
+	if (status->status == 0)
+		status->status = cb_info->cci_rc;
+}
+
+int
+daos_rpc_send_wait(crt_rpc_t *rpc)
```

Comment:
oh, I do not create another task for these management api, which seems overhead for me. Actually I create this just because we do not have to always use task for client side api.